tests/describe_task_queue.go

```diff
@@ -0,0 +1,168 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tests
+
+import (
+	"context"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/require"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/server/common/dynamicconfig"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+type (
+	DescribeTaskQueueSuite struct {
+		*require.Assertions
+		FunctionalTestBase
+	}
+)
+
+func (s *DescribeTaskQueueSuite) SetupSuite() {
+	s.dynamicConfigOverrides = map[dynamicconfig.Key]any{
+		// this is overridden for tests using testWithMatchingBehavior
+		dynamicconfig.MatchingNumTaskqueueReadPartitions.Key():  4,
+		dynamicconfig.MatchingNumTaskqueueWritePartitions.Key(): 4,
+	}
+	s.setupSuite("testdata/es_cluster.yaml")
+}
+
+func (s *DescribeTaskQueueSuite) TearDownSuite() {
+	s.tearDownSuite()
+}
+
+func (s *DescribeTaskQueueSuite) SetupTest() {
+	// Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
+	s.Assertions = require.New(s.T())
+}
+
+func (s *DescribeTaskQueueSuite) TestAddNoTasks_ValidateBacklogInfo() {
+	s.publishConsumeWorkflowTasksValidateBacklogInfo(4, 0)
+}
+
+func (s *DescribeTaskQueueSuite) TestAddSingleTask_ValidateBacklogInfo() {
+	s.publishConsumeWorkflowTasksValidateBacklogInfo(1, 1)
+}
+
+func (s *DescribeTaskQueueSuite) TestAddMultipleTasksMultiplePartitions_ValidateBacklogInfo() {
+	s.publishConsumeWorkflowTasksValidateBacklogInfo(4, 100)
+}
+
+func (s *DescribeTaskQueueSuite) publishConsumeWorkflowTasksValidateBacklogInfo(partitions int, workflows int) {
+	// overriding the ReadPartitions and WritePartitions
+	dc := s.testCluster.host.dcClient
+	dc.OverrideValue(s.T(), dynamicconfig.MatchingNumTaskqueueReadPartitions, partitions)
+	dc.OverrideValue(s.T(), dynamicconfig.MatchingNumTaskqueueWritePartitions, partitions)
```

Comment:
if these are being overridden here, why are they in SetupSuite?

Reply:
Removed `SetupSuite` to override them and only overriding them in `publishConsumeWorkflowTasksValidateBacklogInfo`. Thanks!